### PR TITLE
HdrGen: avoid introducing irritating indentation on  if {} else if {} constructs

### PR DESCRIPTION
**Pull request by UplinkCoder**
before
```
if (a)
{...}
else
    if (b)
    {...}
    else
        if (c)
```
after patch
```
if (a)
{...}
else if (b)
{...}
else if (c)
{...}
```

### CODE REVIEW

File: src/hdrgen.d [hunk 1]
```diff
@@ -322,11 +322,18 @@ public:
...
             buf.writestring("else");
-            buf.writenl();
-            if (!s.elsebody.isScopeStatement())
+            if (!s.elsebody.isIfStatement)
+            {
+                buf.writenl();
+            }
+            else
+            {
+                buf.writeByte(' ');
```
Comment (MartinNowak) [inline src/hdrgen.d]: Line is missing coverage

**Review by MartinNowak — DISMISSED** [review]
Missing test case for the fix.

**Comment by UplinkCoder** [comment]
@MartinNowak addressed.

File: src/hdrgen.d (hunk 1 above)
Comment (UplinkCoder) [inline src/hdrgen.d]: not anymore.

**Comment by UplinkCoder** [comment]
@andralex could you quickly merge this ?

**Comment by andralex** [comment]
auto-merge-squash for the win!

**Comment by UplinkCoder** [comment]
@wilzbach circle ci does not not update the status for gh.
@andralex can you toggle the regular auto-merge ? 
the circle ci status is not updated.

**Comment by wilzbach** [comment]
> @andralex can you toggle the regular auto-merge ?
> the circle ci status is not updated.

That's not the reason.
Auto-merge is blocked due to the "Changes requested" status of this PR. You need to dismiss the review.

> @wilzbach circle ci does not not update the status for gh.

Not sure what's going on there, but CircleCi isn't a "Required" CI atm (unfortunately ... )

**Comment by andralex** [comment]
Let's see how my abuse of power works out :)

**Comment by wilzbach** [comment]
> Let's see how my abuse of power works out :)

Seems like it wasn't enabled for DMD:

```
» 28 Feb 2017 03:04:46.362 186 <190>1 2017-02-28T02:04:46.064707+00:00 app web.1 - - PUT https://api.github.com/repos/dlang/dmd/pulls/6568/merge failed; Method Not Allowed 405.
» 28 Feb 2017 03:04:46.438 252 <190>1 2017-02-28T02:04:46.064725+00:00 app web.1 - - {"message":"Squash merges are not allowed on this repository.","documentation_url":"https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-button"}
```

@andralex: could you please check https://github.com/dlang/dmd/settings?

**Comment by andralex** [comment]
@wilzbach fixed

**Comment by andralex** [comment]
Auto-merge toggled on

**Pull request merged**